pkg/lib/serverTime.js

```diff
@@ -321,13 +316,166 @@
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
+                        else if (fields[0] == "server" || fields[0] == "#server")
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
+                        else if (fields[0] == "sourcedir" || fields[0] == "#sourcedir")
+                            new_text += confline(line, config.sources, fields[1]) + "\n";
+                        else
+                            new_text += line + "\n";
+                    }
+                }
+            }
+
+            const extra = config.extra.filter(e => !!e);
+
+            if (extra.length > 0) {
+                if (!new_text.endsWith("\n"))
+                    new_text += "\n";
+                if (!new_text.endsWith("\n\n"))
+                    new_text += "\n";
+
+                new_text += "# Added by Cockpit\n";
+                for (const e of extra)
+                    new_text += "server " + e + "\n";
+            }
+
+            return new_text;
+        });
+    }
+
+    self.get_ntp_config = function () {
+        /* We support editing the configuration of systemd-timesyncd
+         * or chronyd, by dropping a config file snippet that we
+         * assume full control over.
+         */
+
+        const backend = get_timesync_backend();
+        if (backend == "timesyncd")
+            return get_ntp_config_timesyncd();
+        else if (backend == "chronyd")
+            return get_ntp_config_chronyd();
+        else
+            return Promise.resolve({ supported: false });
+    };
+
+    self.set_ntp_config = function (config) {
+        if (!config || !config.supported)
+            return;
+        if (config.backend == "timesyncd")
+            return set_ntp_config_timesyncd(config);
+        if (config.backend == "chronyd")
+            return set_ntp_config_chronyd(config);
+    };
+
+    self.check_timesync_backend = function () {
+        function wait_enabled(proxy, name) {
+            return new Promise((resolve, reject) => {
+                function check() {
+                    if (proxy.enabled == true) {
+                        proxy.removeEventListener("changed", check);
+                        window.clearTimeout(timeout);
+                        resolve();
+                    }
+                }
+                const timeout = window.setTimeout(() => {
+                    proxy.removeEventListener("changed", check);
+                    reject(cockpit.format(_("Unexpected OS configuration. The $0 has not been enabled. Your changes might not have had any effect."), name));
+                }, 10000);
+                proxy.addEventListener("changed", check);
+                check();
+            });
+        }
+
+        const backend = get_timesync_backend();
+        if (backend == "timesyncd")
+            return wait_enabled(timesyncd_service, "systemd-timesyncd.service");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230113-125341-c8c3ecb6-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)